File: Two Pointer Algo/reverse_string_344.java
class Solution {
    public void reverseString(char[] s) {

        //using two pointer we can solve this questions
        int start = 0, end = s.length-1;
        while( start < end ) {
            char temp = s[start];
            s[start] = s[end];
            s[end] = temp;
            start += 1;
            end -= 1;
        }
        return ;
        
    }
}

File: Two Pointer Algo/reverse_string_344.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: List[str]
        :rtype: None Do not return anything, modify s in-place instead.
        """
        start = 0
        end = len(s)-1
        while start < end:
            temp = s[start]
            s[start] = s[end]
            s[end] = temp
            start  =  start + 1
            end = end - 1
        return
